pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -113,10 +120,17 @@ type GrpcPodConfig struct {
 	// +optional
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
 
-	// RunAsRoot allows admins to indicate that they wish to run the container inside the CatalogSource pod in a privileged
-	// mode as root.  This should only be enabled when running older catalog images which could not be run as non-root.
+	// SecurityContextConfig can be one of "legacy" or "restricted". The grpc pod created is either injected with
+	// the right pod.spec.securityContext and pod.spec.container[*].securityContext values to allow the pod to run in
+	// Pod Security Admission(PSA) "restricted" mode, or does not set these values at all, in which case the pod
+	// can only be run in PSA "privileged" namespaces. By default, SecurityContextConfig is set to "restricted". If the value
+	// is unspecified, the default value of restricted is used.  Specifying any other value will result in a validation error.
+	// When using older catalog images, which could not be run in restricted mode, the SecurityContextConfig should be set to
+	// "legacy". More information about PSA can be found here: https://kubernetes.io/docs/concepts/security/pod-security-admission/
 	// +optional
-	RunAsRoot bool `json:"runAsRoot,omitempty"`
+	// +kubebuilder:validation:Enum=legacy;restricted
+	// +kubebuilder:default:=restricted
```

Comment:
Not positive it matters (or if my suggestion is even correct) since it looks like the CRD generated correctly.
```suggestion
	// +kubebuilder:default=restricted
```

Reply:
According to the [official Kubebuilder Docs](https://book.kubebuilder.io/reference/markers/crd-validation.html#crd-validation), this is correct syntax.